[Elixir] New concept exercise: maps (#1336)
exercism/v3: languages/elixir/exercises/concept/maps/.docs/instructions.md

> In this exercise you're going to write a module to create and manipulate a map or high scores.

Reviewer: ```suggestion
In this exercise you're going to write a module to create and manipulate a map of high scores.
```

> In this exercise you're going to write a module to create and manipulate a map of high scores.

Reviewer: I like this story! Maybe we can leave out the "module" bit (that is an implementation detail, not part of the story) and expand slightly on the story to give it a bit of a flourish?

```suggestion
In this exercise, you're implementing a way to keep track of the high scores for the most popular game in your local arcade hall.
```

Or something like that.

You have 7 functions to implement, all related to returning and manipulating a map of high score data.

### 1. Define a new high score map

> Define the `HighScore.new/0` function which doesn't take any parameters and returns new empty map of highscores.

Reviewer: ```suggestion
Define the `HighScore.new/0` function which doesn't take any parameters and returns a new, empty map of high scores.
```

```elixir
HighScore.new()
# => %{}
```

### 2. Add players to the high score map

> Define `HighScore.add_player/3`, which is a function which takes 3 parameters:

Reviewer: We usually phrase things such that there is also a description of what the function does in the task:

```suggestion
To add a player to the high score map, define `HighScore.add_player/3`, which is a function which takes 3 parameters:
```

Or something similar. This comment also applies to the other tasks.

- The first parameter is the map of scores.
- The second parameter is the name of a player as a string.

> - The third parameter is the score as an integer. The parameter is optional, implement the third parameter with a default value.

Reviewer: Maybe explicitly specify what the default value is?

Author: ```suggestion
- The third parameter is the score as an integer. The parameter is optional, implement the third parameter with a default value of 0.
```
